src/NServiceBus.Testing.Fakes/TestablePipelineContext.cs

```diff
@@ -21,6 +21,11 @@ public TestablePipelineContext(IMessageCreator messageCreator = null)
         }
 
         /// <summary>
+        /// A list of all messages sent with a timeout header.
+        /// </summary>
+        public TimeoutMessage<object>[] TimeoutMessages => timeoutMessages.ToArray();
```

Comment:
I stayed with a field. Having a huge LINQ inside the property accessor does not help in the readability IMHO